conjure-go-client/httpclient/metrics.go

```diff
@@ -154,6 +155,10 @@ func tagStatusFamily(_ *http.Request, resp *http.Response, respErr error) metric
 		return metrics.Tags{metricTagFamily2xx}
 	case resp.StatusCode < 400:
 		return metrics.Tags{metricTagFamily3xx}
+	case resp.StatusCode == 408:
+		return metrics.Tags{metricTagFamilyTimeout}
+	case resp.StatusCode == 429:
```

Comment:
This change makes sense, but I am a little worried about changing the semantic meaning of `4xx` to except 408 and 429. Do we think this could be confusing?

Reply:
We already treat client-side timeouts differently (see https://github.com/palantir/conjure-go-runtime/blob/develop/conjure-go-client/httpclient/metrics.go#L148), but these are not tied to any specific response status code, so not that applicable. I think the value of distinguishing 408s and 429s across other 4xx status codes is significant enough to justify this change. One option to keep both would be to keep 4xx tag, and add Timeout and TooManyRequests as additional tags, this way we don't break anything.